test/template-tests/testcases/deploymentTemplate/apiVersions-Should-Be-Recent.test.ps1

```diff
@@ -39,23 +39,32 @@ foreach ($av in $allApiVersions) { # Then walk over each object containing an Ap
 
     # Now, get the API version as a string
     $apiString = $av.ApiVersion 
-    if ($apiString -like '*-preview') { # If it was a preview API, 
-        # chop off the -preview from the text
-        $apiString = $apiString.Substring(0, $apiString.Length - '-preview'.Length)
-    }
+    $hasDate = $apiString -match "(?<Year>\d{4,4})-(?<Month>\d{2,2})-(?<Day>\d{2,2})"
+    
     
-    $apiDate = $apiString -as [DateTime] # now coerce the apiVersion into a DateTime
-    if (-not $apiDate) {
+    
+    if (-not $hasDate) {
         # If this failed, write an error.
         Write-Error "Api versions must be a fixed date. $FullResourceType is not." -TargetObject $av -ErrorId ApiVersion.Not.Date
         continue
     }
+    $apiDate = [DateTime]::new($matches.Year, $matches.Month, $matches.Day) # now coerce the apiVersion into a DateTime
+
     
 
     # Now find all of the valid versions from this API
-    $validApiVersions = $AllAzureResources.$FullResourceType | 
-        Select-Object -ExpandProperty apiVersions 
-    #! sort this yourself, because it might not be
+    $validApiVersions = # This is made a little tricky by the fact that some resources don't directly have an API version        

```

Comment:
what's the scenario here?  The only case I can think of is when apiProfiles are used, in which case we would just bypass the check on the resource.